Error while checking build/O2/fullCI for a643e97a4ac25c13523acd1b901a33158606c82f at 2024-02-13 17:33:
```
## sw/BUILD/O2-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'


## sw/BUILD/O2Physics-latest/log
/sw/slc8_x86-64/O2/12701-slc8_x86-64-local1/include/Framework/StructToTuple.h:288:29: error: static assertion failed: Too many elements in the struct
/sw/slc8_x86-64/O2/12701-slc8_x86-64-local1/include/Framework/StructToTuple.h:288:29: error: static assertion failed: Too many elements in the struct
/sw/slc8_x86-64/O2/12701-slc8_x86-64-local1/include/Framework/StructToTuple.h:288:29: error: static assertion failed: Too many elements in the struct
/sw/slc8_x86-64/O2/12701-slc8_x86-64-local1/include/Framework/StructToTuple.h:288:29: error: static assertion failed: Too many elements in the struct
/sw/slc8_x86-64/O2/12701-slc8_x86-64-local1/include/Framework/StructToTuple.h:288:29: error: static assertion failed: Too many elements in the struct
/sw/slc8_x86-64/O2/12701-slc8_x86-64-local1/include/Framework/StructToTuple.h:288:29: error: static assertion failed: Too many elements in the struct
/sw/slc8_x86-64/O2/12701-slc8_x86-64-local1/include/Framework/StructToTuple.h:288:29: error: static assertion failed: Too many elements in the struct
/sw/slc8_x86-64/O2/12701-slc8_x86-64-local1/include/Framework/StructToTuple.h:288:29: error: static assertion failed: Too many elements in the struct
/sw/slc8_x86-64/O2/12701-slc8_x86-64-local1/include/Framework/StructToTuple.h:288:29: error: static assertion failed: Too many elements in the struct
/sw/slc8_x86-64/O2/12701-slc8_x86-64-local1/include/Framework/StructToTuple.h:288:29: error: static assertion failed: Too many elements in the struct
/sw/slc8_x86-64/O2/12701-slc8_x86-64-local1/include/Framework/StructToTuple.h:288:29: error: static assertion failed: Too many elements in the struct
/sw/slc8_x86-64/O2/12701-slc8_x86-64-local1/include/Framework/StructToTuple.h:288:29: error: static assertion failed: Too many elements in the struct
/sw/slc8_x86-64/O2/12701-slc8_x86-64-local1/include/Framework/StructToTuple.h:288:29: error: static assertion failed: Too many elements in the struct
/sw/slc8_x86-64/O2/12701-slc8_x86-64-local1/include/Framework/StructToTuple.h:288:29: error: static assertion failed: Too many elements in the struct
/sw/slc8_x86-64/O2/12701-slc8_x86-64-local1/include/Framework/StructToTuple.h:288:29: error: static assertion failed: Too many elements in the struct
/sw/slc8_x86-64/O2/12701-slc8_x86-64-local1/include/Framework/StructToTuple.h:288:29: error: static assertion failed: Too many elements in the struct
/sw/slc8_x86-64/O2/12701-slc8_x86-64-local1/include/Framework/StructToTuple.h:288:29: error: static assertion failed: Too many elements in the struct
/sw/slc8_x86-64/O2/12701-slc8_x86-64-local1/include/Framework/StructToTuple.h:288:29: error: static assertion failed: Too many elements in the struct
/sw/slc8_x86-64/O2/12701-slc8_x86-64-local1/include/Framework/StructToTuple.h:288:29: error: static assertion failed: Too many elements in the struct
/sw/slc8_x86-64/O2/12701-slc8_x86-64-local1/include/Framework/StructToTuple.h:288:29: error: static assertion failed: Too many elements in the struct
/sw/slc8_x86-64/O2/12701-slc8_x86-64-local1/include/Framework/StructToTuple.h:288:29: error: static assertion failed: Too many elements in the struct
/sw/slc8_x86-64/O2/12701-slc8_x86-64-local1/include/Framework/StructToTuple.h:288:29: error: static assertion failed: Too many elements in the struct
/sw/slc8_x86-64/O2/12701-slc8_x86-64-local1/include/Framework/StructToTuple.h:288:29: error: static assertion failed: Too many elements in the struct
/sw/slc8_x86-64/O2/12701-slc8_x86-64-local1/include/Framework/StructToTuple.h:288:29: error: static assertion failed: Too many elements in the struct
/sw/slc8_x86-64/O2/12701-slc8_x86-64-local1/include/Framework/StructToTuple.h:288:29: error: static assertion failed: Too many elements in the struct
/sw/slc8_x86-64/O2/12701-slc8_x86-64-local1/include/Framework/StructToTuple.h:288:29: error: static assertion failed: Too many elements in the struct
/sw/slc8_x86-64/O2/12701-slc8_x86-64-local1/include/Framework/StructToTuple.h:288:29: error: static assertion failed: Too many elements in the struct
/sw/slc8_x86-64/O2/12701-slc8_x86-64-local1/include/Framework/StructToTuple.h:288:29: error: static assertion failed: Too many elements in the struct
/sw/slc8_x86-64/O2/12701-slc8_x86-64-local1/include/Framework/StructToTuple.h:288:29: error: static assertion failed: Too many elements in the struct
/sw/slc8_x86-64/O2/12701-slc8_x86-64-local1/include/Framework/StructToTuple.h:288:29: error: static assertion failed: Too many elements in the struct
/sw/slc8_x86-64/O2/12701-slc8_x86-64-local1/include/Framework/StructToTuple.h:288:29: error: static assertion failed: Too many elements in the struct
/sw/slc8_x86-64/O2/12701-slc8_x86-64-local1/include/Framework/StructToTuple.h:288:29: error: static assertion failed: Too many elements in the struct
/sw/slc8_x86-64/O2/12701-slc8_x86-64-local1/include/Framework/StructToTuple.h:288:29: error: static assertion failed: Too many elements in the struct
/sw/slc8_x86-64/O2/12701-slc8_x86-64-local1/include/Framework/StructToTuple.h:288:29: error: static assertion failed: Too many elements in the struct
/sw/slc8_x86-64/O2/12701-slc8_x86-64-local1/include/Framework/StructToTuple.h:288:29: error: static assertion failed: Too many elements in the struct
/sw/slc8_x86-64/O2/12701-slc8_x86-64-local1/include/Framework/StructToTuple.h:288:29: error: static assertion failed: Too many elements in the struct
/sw/slc8_x86-64/O2/12701-slc8_x86-64-local1/include/Framework/StructToTuple.h:288:29: error: static assertion failed: Too many elements in the struct
/sw/slc8_x86-64/O2/12701-slc8_x86-64-local1/include/Framework/StructToTuple.h:288:29: error: static assertion failed: Too many elements in the struct
/sw/slc8_x86-64/O2/12701-slc8_x86-64-local1/include/Framework/StructToTuple.h:288:29: error: static assertion failed: Too many elements in the struct
/sw/slc8_x86-64/O2/12701-slc8_x86-64-local1/include/Framework/StructToTuple.h:288:29: error: static assertion failed: Too many elements in the struct
/sw/slc8_x86-64/O2/12701-slc8_x86-64-local1/include/Framework/StructToTuple.h:288:29: error: static assertion failed: Too many elements in the struct
/sw/slc8_x86-64/O2/12701-slc8_x86-64-local1/include/Framework/StructToTuple.h:288:29: error: static assertion failed: Too many elements in the struct
/sw/slc8_x86-64/O2/12701-slc8_x86-64-local1/include/Framework/StructToTuple.h:288:29: error: static assertion failed: Too many elements in the struct
[0 more errors; see full log]
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/12701/a643e97a4ac25c13523acd1b901a33158606c82f/build_O2_fullCI/pretty.html).

--- a/Framework/Foundation/include/Framework/StructToTuple.h
+++ b/Framework/Foundation/include/Framework/StructToTuple.h
@@ -135,10 +135,12 @@ struct UniversalType {
 template <typename T>
 consteval auto brace_constructible_size(auto... Members)
 {
-  if constexpr (requires { T{Members...}; } == false)
+  if constexpr (requires { T{Members...}; } == false) {
+    static_assert(sizeof...(Members) != 0, "You need to make sure that you have implicit constructors or that you call the explicit constructor correctly.");
     return sizeof...(Members) - 1;
-  else
+  } else {
     return brace_constructible_size<T>(Members..., UniversalType{});
+  }
 }
 #else
 template <typename T>

